src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -227,19 +227,6 @@ namespace {
 using EventEngine = ::grpc_event_engine::experimental::EventEngine;
 using TaskHandle = ::grpc_event_engine::experimental::EventEngine::TaskHandle;
 
-grpc_core::CallTracerAnnotationInterface* CallTracerIfSampled(
-    grpc_chttp2_stream* s) {
-  if (!grpc_core::IsTraceRecordCallopsEnabled()) {
-    return nullptr;
-  }
-  auto* call_tracer =
-      s->arena->GetContext<grpc_core::CallTracerAnnotationInterface>();
-  if (call_tracer == nullptr || !call_tracer->IsSampled()) {
```

Comment:
It looks like you're removing this check of `IsSampled()`.  That seems like a regression.

Reply:
This check doesn't bring us much. The implementations of `RecordAnnotation()` already check whether the tracer is sampled.

Reply:
We are constructing an `HttpAnnotation` object which would end up copying some fields over. Alright, added `IsSampled()` checks at the relevant locations